utils/unittest/epoch_builder.go

```diff
@@ -391,31 +392,25 @@ func (builder *EpochBuilder) addBlock(block *flow.Block) {
 // NOTE: This func should only be used after BuildEpoch to extend the commit phase
 func (builder *EpochBuilder) AddBlocksWithSeals(n int, counter uint64) *EpochBuilder {
 	for i := 0; i < n; i++ {
-		// given the last 3 blocks in state a <- b <- c
-		// the new block will contain
-		// - seal for c
-		// - execution receipt for b
-		// - execution result for a
-		a := builder.blocks[len(builder.blocks)-1]
-		b := builder.blocks[len(builder.blocks)-2]
-
-		// get execution result for third highest block
-		//builder.blocks[len(builder.blocks)-3].Payload.Receipts
-
-		// get receipt for the highest block
-		receiptF := ReceiptForBlockFixture(a)
-
-		// build a new block
-		// this new block contains a seal for previous highest block and a receipt for the highest block
-		block := BlockWithParentFixture(a.Header)
+		// Given the last 2 blocks in state A <- B when we add block C it will contain the following.
+		// - seal for B
+		// - execution result for A
+		a := builder.blocks[len(builder.blocks)-2]
+		b := builder.blocks[len(builder.blocks)-1]
+
+		receiptB := ReceiptForBlockFixture(b)
+
+		block := BlockWithParentFixture(b.Header)
 		seal := Seal.Fixture(
-			Seal.WithResult(b.Payload.Results[0]),
+			Seal.WithResult(a.Payload.Results[0]),
+		)
+
+		payload := PayloadFixture(
+			WithReceipts(receiptB),
+			WithExecutionResults(&receiptB.ExecutionResult),
```

Comment:
```suggestion
			WithReceipts(receiptB),
```
When we add a receipt, by default it will include the result (since the result is an embedded field of a receipt)

Reply:
https://github.com/onflow/flow-go/pull/1940/commits/0149bd876a34803334e4b10e3873b68e344a46e3